### ISSUE TITLE
H264 (MP4) videos can't play

### ISSUE THREAD

**Issue by alarixnia** [rmottola/Arctic-Fox#202]
Hello, I'd like to use the Indivious or ViewTube frontends for YouTube with ArcticFox. Currently, these sites fail to play videos because they try to load a h264 video and it fails.

Is there any way I can build Arcticfox with H264 support enabled? I'd like to distribute these builds with the NetBSD package.

Thanks.

**Comment by alarixnia** [comment]
Related invidious issues:

https://github.com/iv-org/invidious/issues/3993
https://github.com/iv-org/invidious/issues/3677

**Comment by rmottola** [comment]
To be honest, I don't know if there is an option not enabled or still some code path missing.
Do you know what component should provide h264 decoding? ffmpeg?

**Comment by rmottola** [comment]
Are you able to use h264 for wha tyou need on latest firefox?

I am on linux with latest and testing https://canonical.github.io/inbrowser-encode-test/ only works for vp8, but that is encoding

**Comment by rmottola** [comment]
https://tools.woolyss.com/html5-audio-video-tester/

this confirms that SeaMonkey and Firefox can play 264, but ArcticFox does not. At least we have an easy test.

**Comment by alarixnia** [comment]
Yeah I believe the h264 support should be provided through ffmpeg.

**Comment by rmottola** [comment]
> Yeah I believe the h264 support should be provided through ffmpeg.

hmm.. Just by chance do you know if FF52 ESR has h264 support? and FF 45ESR? that way I have targets to compare. ArcticFox has some PaleMoon deriving ffmpeg updates.

**Comment by alarixnia** [comment]
it appears my memory is playing tricks on me, I thought firefox52 had h264 support on unix.

https://html5test.com/ reports no.

it's built against ffmpeg.

**Comment by rmottola** [comment]
Then it will be quite a bit more work and won't get mad over it right now. First goal is to match FF 52 as best as possible.
If you have FF52 handy, you can use my proposed test to have more granular checks.  https://tools.woolyss.com/html5-audio-video-tester/

according to https://caniuse.com/mpeg4  support was provided by gstreamer, which became obsolete and removed. So I don't know when it added different support.

According to https://caniuse.com/mpeg4 it was for WebRTC only back then.

**Comment by rmottola** [comment]
Hmm.. more complicated. Read here:
https://askubuntu.com/questions/389437/how-do-i-get-html5-h-264-video-working-on-firefox

and I now test on Linux firefox 125 and it has an addon installed for H264. 
Seamoneky has not! Wonder...

I guess ArcticFox doesn't get this ad on and FF52 neither.
I suppose there is no easy way, especially on BSD?

**Comment by alarixnia** [comment]
We don't need an addon for h264 on BSD, it's just ffmpeg compiled with h264 support enabled as far as I know.

**Comment by alarixnia** [comment]
I saw that tenfourfox supports h264 through ffmpeg: https://github.com/classilla/tenfourfox/wiki/PlayingH.264

**Comment by rmottola** [comment]
It has some very custom plugin and libraries, apparently. Or do seamonkey and firefox also use ffmpeg? This needs a lot of investigation.

**Comment by alarixnia** [comment]
Seamonkey and firefox both just need ffmpeg in recent versions.

**Comment by rmottola** [comment]
> Seamonkey and firefox both just need ffmpeg in recent versions.

I noticed that Firefox between v52 and v60 added ffmpeg v58. Can you see if FF60 works for you where you already said 52 didn't?
On the other side, AF had 58, since it came from PaleMoon which it backported it. There was a typo in loading versions, I fixed it, but it does not improve the situation. I tried to import quite some media related code, but no help. If you want to try building dev branch, try yourself.

**Comment by rmottola** [comment]
@alarixnia hmm.. Current dev branch has H264 working on MacOS, but same code does not enable it on Linux or NetBSD.

I wonder if WebRTC has to be enabled...

**Comment by the-r3dacted** [comment]
roytam1's New Moon 27 plays H.264 videos that Arctic Fox can't play. Although I'm testing on Windows, idk if the code could work on Mac or Linux.

**Comment by rmottola** [comment]
@K4sum1 that is not test.. I don't know if @roytam1 changed something or it just works on windows. As I wrote, H264 works on Mac...

**Comment by roytam1** [comment]
> I don't know if @roytam1 changed something or it just works on windows.

yeah I did change it to load external libav(LAV Filter) decoder to do decoding.
